encryption/src/main/java/org/apache/solr/encryption/EncryptionRequestHandler.java

```diff
@@ -260,12 +339,133 @@ public void handleRequestBody(SolrQueryRequest req, SolrQueryResponse rsp) throw
+    boolean success = false;
+    String collectionName = null;
+    State collectionState = null;
+    long timeAllowedMs = req.getParams().getLong(TIME_ALLOWED, 0);
+    long maxTimeNs = timeAllowedMs <= 0 ? Long.MAX_VALUE : startTimeNs + timeAllowedMs;
+    try {
+      collectionName = req.getCore().getCoreDescriptor().getCollectionName();
+      if (collectionName == null) {
+        throw new SolrException(SolrException.ErrorCode.SERVER_ERROR, "Parameter " + DISTRIB + " can only be used in Solr Cloud mode.");
+      }
+      log.debug("Encrypt request distributed for keyId={} collection={}", keyId, collectionName);
+      DocCollection docCollection = req.getCore().getCoreContainer().getZkController().getZkStateReader().getCollection(collectionName);
+      if (docCollection == null) {
+        throw new SolrException(SolrException.ErrorCode.SERVER_ERROR, "Parameter " + DISTRIB + " present but collection '" + collectionName + "' not found.");
+      }
+      try (SolrClientHolder solrClient = getHttpSolrClient(req)) {
+        ModifiableSolrParams params = createDistributedRequestParams(req, rsp, keyId);
+        Collection<Slice> slices = docCollection.getActiveSlices();
+        Collection<Callable<State>> encryptRequests = new ArrayList<>(slices.size());
+        final String collectionNameFinal = collectionName;
+        for (Slice slice : slices) {
+          Replica replica = slice.getLeader();
+          if (replica == null) {
+            log.error("No leader found for shard {}", slice.getName());
+            collectionState = State.ERROR;
+            continue;
+          }
+          encryptRequests.add(() -> sendEncryptionRequestWithRetry(replica, params, solrClient.getClient(), keyId, collectionNameFinal));
+        }
+        try {
+          List<Future<State>> responses = timeAllowedMs <= 0 ?
+              executor.invokeAll(encryptRequests)
+              : executor.invokeAll(encryptRequests, timeAllowedMs, TimeUnit.MILLISECONDS);
+          if (isTimeout(maxTimeNs)) {
+            log.warn("Timeout distributing encryption request for keyId={} collection={}", keyId, collectionName);
+            if (collectionState == null || State.TIMEOUT.priority > collectionState.priority) {
+              collectionState = State.TIMEOUT;
+            }
+          }
+          for (Future<State> response : responses) {
+            State state;
+            try {
+              state = response.get();
+            } catch (ExecutionException e) {
+              collectionState = State.ERROR;
+              break;
+            }
+            if (collectionState == null || state.priority > collectionState.priority) {
+              collectionState = state;
+            }
+          }
+        } catch (InterruptedException e) {
+          collectionState = State.INTERRUPTED;
+        }
+        success = collectionState == null || collectionState.isSuccess();
```

Comment:
why is null a success?

Reply:
null only happens if there is no distribution because there are no slices. I don't know if it can happen actually, but the request to encrypt all shards is complete.